frontend/src/API/SiteAPIHelper.ts

```diff
@@ -29,8 +29,23 @@ export default class SiteAPIHelper {
         }
     }
 
+    async subscriptons(): Promise<SiteWithUserInfo[]> {
+        const result = await this.api.subscriptions();
+        this.appState.setSubscriptions(result.subscriptions);
+        return result.subscriptions;
+    }
+
     async list(page: number, perpage: number): Promise<SiteWithUserInfo[]> {
         const result = await this.api.list(page, perpage);
         return result.sites;
     }
+
+    async updateSiteInfo() {
+        if (this.appState.site) {
+            const result = await this.api.site(this.appState.site);
+            if (result && result.site.site === this.appState.site) {
```

Comment:
`result.site.site === this.appState.site`
is this check needed? It's getting site info by a site name so not quite clear when some other site could be returned here

Reply:
`this.api.site` is an async call. While it's in progress, `this.appState.site` could have been changed.